src/gui/entry/EntryModel.cpp

```diff
@@ -145,17 +220,148 @@ QVariant EntryModel::data(const QModelIndex& index, int role) const
             }
             return result;
         case Username:
-            result = entry->resolveMultiplePlaceholders(entry->username());
+            /*
+             * Display usernames hidden or visible according to current state
+             * of 'Hide Usernames' setting
+             */
+            if (m_hideUsernames) {
+                result = EntryModel::HiddenContentDisplay;
+            }
+            else {
+                //result = entry->username();
+                result = entry->resolveMultiplePlaceholders(entry->username());
+            }
             if (attr->isReference(EntryAttributes::UserNameKey)) {
                 result.prepend(tr("Ref: ","Reference abbreviation"));
             }
             return result;
+        case Password:
+            /*
+             * Display passwords hidden or visible according to current state
+             * of 'Hide Passwords' setting
+             */
+            if (m_hidePasswords) {
+                result = EntryModel::HiddenContentDisplay;
+            }
+            else {
+                //result = entry->resolveMultiplePlaceholders(entry->password());
+                result = entry->password();
+            }
+            if (attr->isReference(EntryAttributes::PasswordKey)) {
+                result.prepend(tr("Ref: ","Reference abbreviation"));
+            }
+            return result;
         case Url:
+            //result = entry->resolveMultiplePlaceholders(entry->displayUrl());
             result = entry->displayUrl();
             if (attr->isReference(EntryAttributes::URLKey)) {
                 result.prepend(tr("Ref: ","Reference abbreviation"));
             }
             return result;
+        case Notes:
+            /*
+             * Display only first line of notes in simplified format like
+             * KeePassX does
```

Comment:
?

Reply:
I don't think we need to refer back to KeePassX. I'm not entirely certain about its future. :wink:

Reply:
Well, I don't know about you, but I consider KeePassX 0.4.x a great application, while the rewrite was... well... I don't think I can say this nicely. More respect for the classic ;)

Reply:
The moment this gets merged, this comment loses its meaning, because then there is no difference between KeePassXC and KeePassX anymore. I'm just a little concerned about polluting the code with historic comments which will have no meaning to a developer tomorrow.